airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -401,7 +402,11 @@ private SyncCheckConnectionFailure checkConnections(final GenerateInputActivity.
     final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(sourceConfig);
     final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
 
-    if (isResetJob(sourceLauncherConfig) || checkFailure.isFailed()) {
+    final JobCheckFailureInput jobStateInput =
+        new JobCheckFailureInput(Long.parseLong(jobRunConfig.getJobId()), jobRunConfig.getAttemptId().intValue(), connectionId);
+    var isLastJobOrAttemptFailure = jobCreationAndStatusUpdateActivity.isLastJobOrAttemptFailure(jobStateInput);
```

Comment:
You need to create a new workflow version here. Otherwise it will make the running workflow to fail when this is being deploy. The workflow version is a temporal internal functionality that ensure compatibility between an old running version of this workflow and the new one you are trying to introduce. You canfind an example of the usage of such a version on line 287 in the report success function. I will be happy to pair on this if needed.

Another issue here is that you need to wrap your activity code in the `runMandatoryActivityWithOutput` function. This is needed to ensure that we don't block the workflow is this activity fails for any reason.

Reply:
I believe that for the version comment the test `WorkflowReplayingTest` should fail as of now.

Reply:
@benmoriceau, I added the changes you posted above. But I'm not sure if I did it right because I failed to update the version correctly. Please review my pull request again and if I'm doing something wrong, tell me what I should add and change. (I'm working with temporal for the first time, and I may be wrong in something)